app/logic/events.py

```diff
@@ -146,10 +147,19 @@ def getTrainingEvents(term):
         with the most programs (highest count) by doing this we can ensure that the event being
         returned is the All Trainings Event.
     """
-    trainingEvents = (Event.select(Event)
-                           .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
-                           .where(Event.isTraining,
-                                  Event.term == term))
+    if (g.current_user.isStudent and not g.current_user.isBonnerScholar) or g.current_user.isFaculty:
+        trainingEvents = (Event.select(Event)
+                                .join(ProgramEvent)
+                                .join(Program)
+                                .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
+                                .where(Event.isTraining,
+                                        Event.term == term,
+                                        not Program.isBonnerScholars))
+    else:
+        trainingEvents = (Event.select(Event)
+                               .order_by(Event.isAllVolunteerTraining.desc(), Event.startDate)
+                               .where(Event.isTraining,
+                                      Event.term == term))
```

Comment:
This definitely breaks the test (and the new functionality isn't tested either). We shouldn't use the global variable inside this function. if the training events now depend on the user, we should pass a user into the function